Solidus 3 preparation

cpfergus1 opened the pull request.

SamuelMartini committed:
Use BigDecimal() instead of BigDecimal.new()

SamuelMartini committed:
Use update instead of update_attributes

SamuelMartini committed:
Avoid UnknownAttributeReference exception

From Rails 5.2 is not possible to pass a raw string to order.

SamuelMartini committed:
Fix delete sale price feature spec

SamuelMartini committed:
Use discard gem api

https://github.com/solidusio/solidus/pull/3488

cpfergus1 committed:
introduce a notice for removal of acts_as_paranoid

Once 2.10 is no longer supported, act as paranoid will be removed and
destroy will no longer work as it previously did.

cpfergus1 committed:
Implemented price finding method to bridge Solidus version functionality gap

This is required to merge the functionality of price_in and price_for_options.
The functionality of finding prices changes in solidus 2.11, 3.0 and master
so this function serves to work for all three for these scenarios

cpfergus1 committed:
Fixed deprecated `EngineExtensions::Decorators` module reference

cpfergus1 committed:
Bump version support to Solidus 3.0 and Ruby >=2.5

cpfergus1 committed:
Reload after factory generation

In Solidus 3.1, the test factory no longer appends the prices to the variant
without the object being reloaded. Implemented reload to assist specs to
pass.

cpfergus1 committed:
Update test to use literal

The sales_price is made from an price associated with the variant. To verify
that the sales_price recieved the properties of the original price,
you would want to test against the original price properties. In 3.1,
this test failed due to the variant having more than one price, a master
price and a test set price. The previous behavior would ask for the last
price, `variant.prices.last.currency`, assuming a new price was generated
when it was actually pulling the master variant price which was in "USD".
In tests for 3.0 and below, the first price is also the last price,
which is why it never failed in the first place. Swapped to literal to
avoid the error.

cpfergus1 committed:
Use let in lieu of instance variables

cpfergus1 committed:
Updated factories loading after recent changes in solidus

This commit reflects changes made by Solidus [PR #169]
(solidusio/solidus_dev_support#169) When testing extensions
using Solidus vesions that support the definition_file_paths,
it will load the core factory first and then the ones defined in
`lib/extension_name/testing_support/factories`.

cpfergus1 committed:
Increased tolerance of test

Test was failing within a couple ten thousandths of a second.
Increased by one second to accommodate float.

QuintinAdam commented on solidus_sale_prices.gemspec:
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |s|
   end
 
-  s.required_ruby_version = '~> 2.4'
+  s.required_ruby_version = '~> 2.5'
```
Could we change this to allow Ruby 3 to be used as well? 
```suggestion
  s.required_ruby_version = Gem::Requirement.new('>= 2.5')
```

cpfergus1 committed:
Remove non-required Rails version requirement

The bug that this code fixed has been patched in the the latest versions
of bundler and is no longer required.
